### PR TITLE
Smitha Hosmani

### PR DESCRIPTION
**Pull request by hsmitha26**

### CODE REVIEW

File: lib/pantry.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  def stock_check(ingredient_object)
+    @stock[ingredient_object]
+  end
+
+  def restock(ingredient_object, amount)
+    @stock[ingredient_object] += amount
+  end
+
+  def enough_ingredients_for?(recipe_object)
+    recipe_object.ingredients_required.all? {|ingredient_object, amount| @stock[ingredient_object] >= amount}
```
Comment (BrianZanti) [inline lib/pantry.rb]: Great job! I think `all?` was the best enumerable for the job 👏 I think you should use the multiline syntax here. This is a lot for one line. Also, notice how you repeated the code `@stock[ingredient_object]` code here and in your `stock_check` method. In this case, you should reuse your `stock_check` method.

File: lib/recipe.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  def amount_required(ingredient_object)
+    @ingredients_required[ingredient_object]
+  end
+
+  def ingredients
+    @ingredients_required.keys
+  end
+
+  def total_calories
+    @ingredients_required.sum {|ingredient_object, amount| ingredient_object.calories * amount}
```
Comment (BrianZanti) [inline lib/recipe.rb]: Great job picking `sum`!

File: test/cook_book_test.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  def test_it_can_list_summary_of_recipes
+    @mac_and_cheese.add_ingredient(@cheese, 2)
+    @mac_and_cheese.add_ingredient(@mac, 8)
+    @cookbook.add_recipe(@mac_and_cheese)
+
+    @burger.add_ingredient(@ground_beef, 4)
+    @burger.add_ingredient(@bun, 100)
+    @cookbook.add_recipe(@burger)
+
+    expected = [{:name=>"Mac and Cheese",
```
Comment (BrianZanti) [inline test/cook_book_test.rb]: Good job breaking this out onto multiple lines. I think you could've broken it out even further.

File: test/pantry_test.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  def test_it_has_enough_ingredients_to_make_recipe
+    @mac_and_cheese.add_ingredient(@cheese, 2)
+    @mac_and_cheese.add_ingredient(@mac, 8)
+    @pantry.restock(@cheese, 5)
+    @pantry.restock(@cheese, 10)
+    assert_equal false, @pantry.enough_ingredients_for?(@mac_and_cheese)
+
+    @pantry.restock(@mac, 5)
```
Comment (BrianZanti) [inline test/pantry_test.rb]: This is a good edge case to add

File: test/recipe_test.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  end
+
+  def test_it_starts_without_any_ingredients
+    assert_equal ({}), @mac_and_cheese.ingredients_required
+  end
+
+  def test_it_can_add_ingredients
+    @mac_and_cheese.add_ingredient(@cheese, 2)
+    @mac_and_cheese.add_ingredient(@mac, 8)
+    expected = {
```
Comment (BrianZanti) [inline test/recipe_test.rb]: Great job with this expected value!

**Comment by BrianZanti** [comment]
Great job Smitha! You completed iteration 3 🎉
